src/node/rpc/member_frontend.h

```diff
@@ -899,8 +899,9 @@ namespace ccf
         js::Runtime rt;
         js::Context context(rt, js::TxAccess::GOV_RO);
         rt.add_ccf_classdefs();
+        js::TxContext txctx{&ctx.tx};
         js::populate_global(
-          nullptr,
+          &txctx,
```

Comment:
Writing this here so we have a record on the PR:

I was expecting we'd need to add more changes here, so that validation is only run once, when a proposal is first submitted. If it is run multiple times, and transactionally against new states of the KV, it becomes a lot easier to get validation errors during a later execution (ie - assert there are <=3 members, this is true when the proposal is submitted but not when the 2nd vote arrives), which is ugly as we can't report/handle these late validation errors nicely.

It looks like that change is unneeded, because the validation is _already_ run only once, when the proposal is created. We call `resolve_proposal()` in both `POST /proposals` and `POST /proposals/{id}/ballots`, but evaluate the `"validate"` function only in `POST /proposals`. @achamayou can you confirm that understanding?